### ISSUE TITLE
neo4j 4.2.3 installation

### ISSUE THREAD

**Issue by winternewt** [saezlab/CKG-BioCypher#9]
In https://ckg.readthedocs.io/en/latest/intro/getting-started-with-neo4j.html

For me, on Ubuntu 22:
apt-get install neo4j=1:4.2.3 pulls one of the newer "cypher-shell" packages, which in turn pulls openjdk-17
Solved by specifying exact cypher-shell version matching the neo4j
sudo apt-get install neo4j=1:4.2.3 cypher-shell=1:4.2.3

**Comment by antonkulaga** [comment]
I confirm, neo4j 4.x is a biocypher nightmare, most of my dependency hell issues were connected with the fact that you use 4.x version instead of 5.x and also have it in transient dependencies

**Comment by winternewt** [comment]
Apologies, these issues actually belong to https://github.com/MannLabs/CKG/tree/master which is apparently non maintained. I'll try to come up with a set of scripts and a docker-compose to lower the entry threshold for this, because the prerequisite of "Data
CKG Neo4j database dump (available [here](https://data.mendeley.com/datasets/mrcf7f4tc2/3))" turn out remarkably hard to reconstruct.

**Comment by slobentanzer** [comment]
Yes, this (reproducing the CKG) is a maintenance issue first and foremost. Also one of the reasons we developed the framework in the first place. This CKG version here is a relatively early biocypher pipeline, and as the major lesson, I would not recommend streaming from one Neo4j DB into another if it can be avoided.

@antonkulaga I am not sure I get your point. Who/what do you mean with "you"?

@winternewt What is the issue with loading the dump? Neo4j 4.x should work fine if you just load the file you can download at the given link.

**Comment by antonkulaga** [comment]
>I am not sure I get your point. Who/what do you mean with "you"?

I mean that biocypher uses neo4j 4.x, it is also in transient poetry dependencies as well, so switching to 5.x is somewhat a pain.

**Comment by slobentanzer** [comment]
Ah, you are referring to the Bolt driver Python package. This issue is about the Neo4j software itself, so only loosely related.

We already have plans to remove the dependency on neo4j-utils, as it turned out to be ineffective to separate them. Issue here: https://github.com/biocypher/biocypher/issues/332

@nilskre FYI

**Comment by winternewt** [comment]
> @winternewt What is the issue with loading the dump? Neo4j 4.x should work fine if you just load the file you can download at the given link.

There are numerous environment issues with setting Neo4j 4.x with this dump using the CKG docs, see the issue by mistake opened in this repo for reference:
https://github.com/biocypher/clinical-knowledge-graph/issues/10
The db is named 'graph.db' (it was probably migrated from v3) not matching the v4 default naming (neo4j) etc etc etc. Took me a few hours to sort them all out.

I propose the pull eliminating all those, thus ensuring quick reproducibility, @slobentanzer kindly check
https://github.com/biocypher/clinical-knowledge-graph/pull/13

**Comment by slobentanzer** [comment]
Reads like a lesson in reproducibility (of which we unfortunately have many) 🥲

Many thanks for opening a PR! Will check it out ASAP
